app/ante/evm/fee_checker.go

```diff
@@ -36,32 +36,43 @@ func NewDynamicFeeChecker(k DynamicFeeEVMKeeper) authante.TxFeeChecker {
 			return checkTxFeeWithValidatorMinGasPrices(ctx, feeTx)
 		}
 
-		return feeChecker(ctx, k, feeTx)
+		return feeChecker(ctx, fmk, feeTx)
 	}
 }
 
 // feeChecker returns the effective fee and priority for a given transaction.
 func feeChecker(
 	ctx sdk.Context,
-	k DynamicFeeEVMKeeper,
+	k FeeMarketKeeper,
 	feeTx sdk.FeeTx,
 ) (sdk.Coins, int64, error) {
 	denom := config.GetEVMCoinDenom()
 	ethConfig := config.GetChainConfig()
-	baseFee := k.GetBaseFee(ctx, ethConfig)
-	if baseFee == nil {
+	baseFee := k.GetBaseFee(ctx)
+
+	// if baseFee is nil because it is disabled
+	// or not found, consider it as 0
+	// so the DynamicFeeTx logic can be applied
+	if baseFee.IsNil() {
+		baseFee = sdkmath.LegacyZeroDec()
+	}
```

Comment:
nit: we can move getting the baseFee from state after the IsLondon check, 'cause in case it is not london, will not need the baseFee